crates/telio-lana/src/lib.rs

```diff
@@ -62,26 +68,39 @@ pub fn init_lana(
     app_version: String,
     prod: bool,
 ) -> Result<moose::Result, moose::Error> {
+    let (tx, rx) = sync_channel(1);
+
     if !is_lana_initialized() {
-        let result = moose::init(
+        if let Err(err) = moose::init(
             event_path,
             LANA_APP_NAME.to_string(),
             app_version,
             LANA_MOOSE_VERSION.to_string(),
             prod,
-            Box::new(moose_callbacks::MooseCallbacks),
-            Box::new(moose_callbacks::MooseCallbacks),
-        );
-        if let Some(error) = result.as_ref().err() {
-            telio_log_warn!("[Moose] Error: {} on call to `{}`", error, "init");
-        }
+            Box::new(moose_callbacks::MooseInitCallback { init_tx: tx }),
+            Box::new(moose_callbacks::MooseErrorCallback),
+        ) {
+            telio_log_warn!("[Moose] Error: {} on call to `{}`", err, "moose_init");
+            return Err(err);
+        };
 
-        if result.is_ok() {
-            MOOSE_INITIALIZED.store(true, DEFAULT_ORDERING);
-            init_context_info();
+        match rx.recv_timeout(Duration::from_secs(2)) {
```

Comment:
please make a named constant out of that duration, maybe something like `MOOSE_MAX_INIT_TIME` or something similar to make it clear what is it's purpose.

Also, how was this 2s value selected? Do we know what is the typical time it takes on low(est) powered devices we support?

Reply:
Agree. I came up with this magic number of 2s inspired by the libdrop implementation, which seemed reasonable.

> Do we know what is the typical time it takes on low(est) powered devices we support?
 I have no idea 🤔

Reply:
Maybe moose devs know something about it and/or there are some metrics that track that? Additionally you could consider doing something like reducing your cpu frequency to the lowest possible value and measure it yourself in such conditions (I've mentioned how to do this on linux in the comment of LLT-3626).

Reply:
Limited cpu/stressed and the values were always between 10-20ms.